getconf.sh

```diff
@@ -2,11 +2,13 @@
 
 ifdef() {
 	printf 'static const struct var %s[] = {\n' "$1"
-	awk '{printf("#ifdef %s\n\t{\"%s\",\t%s},\n#endif\n", $2, $1, $2)}'
+	while read -r def func _; do
+		printf '#ifdef %s\n\t{"%s",\t%s},\n#endif\n' "$func" "$def" "$func"
+	done
```

Comment:
The other change is fine, but I don't see the point of this change. In my opinion it doesn't make things simpler.

awk is a POSIX standard tool, so is available on all platforms we support.

Reply:
Sure, but why not allow the script to work in any situation with a functioning shell, which is already needed? I know printf isn't necessarily builtin, but any shell expected to work has access to it.

Reply:
Have you run into a situation where awk isn't available? I think if you tried to run a system without it, you'd hit all sorts of other issues running various scripts.

Reply:
Anyway, feel free to submit both changes to the mailing list (preferably as separate patches). I can apply the cat change, and we can see if anyone else has an opinion about the awk -> printf change. I will apply it if there is some support for it.